src/engine/Operation.cpp

```diff
@@ -213,17 +214,16 @@ void Operation::checkAbortion() const {
 
 // ______________________________________________________________________
 
-void Operation::checkAbortion(const auto& detailSupplier) const {
+void Operation::checkAbortion(const std::invocable auto& detailSupplier) const {
   abortionHandle_->throwIfAborted(detailSupplier);
 }
 
 // ______________________________________________________________________
 
 std::chrono::seconds Operation::remainingTime() const {
   auto interval = deadline_ - std::chrono::steady_clock::now();
-  return interval >= std::chrono::seconds::zero()
-             ? std::chrono::duration_cast<std::chrono::seconds>(interval)
-             : std::chrono::seconds::zero();
+  return std::max(std::chrono::seconds::zero(),
+                  std::chrono::duration_cast<std::chrono::seconds>(interval));
```

Comment:
Is it sensible to use `seconds` (which is integral) here, 
so would `0.3 seconds` be cast to `0 seconds` and then sometimes trigger timeout problems?
(I know that this is currently only used for the checks for non-cancellable sorting functions, but still it seems like an eternity. Maybe just use `milliseconds`.